### ISSUE TITLE
[NOT A BUG] sktime.forecasting.pytorchforecasting returning zero for forecasts

### ISSUE THREAD

**Issue by jobs-git** [sktime/sktime#8261]
**Describe the bug**
sktime.forecasting.pytorchforecasting returning zero for forecasts even if following line per line the example provided in sktime: https://github.com/sktime/sktime/blob/main/examples/01c_forecasting_hierarchical_global.ipynb

However, on the test, there appears to be a result for y_pred, but when testing, it returns zero.

**To Reproduce**

See this notebook: https://colab.research.google.com/drive/1xVB9HGVtXm24x4CQXduNMwBegvg6PrX7?usp=sharing

```python
from sklearn.model_selection import train_test_split

from sktime.utils._testing.hierarchical import _make_hierarchical

data = _make_hierarchical(
    hierarchy_levels=(100, 1), max_timepoints=10, min_timepoints=10, n_columns=1
)
data = data.droplevel(1)
y_train, y_test = train_test_split(data, test_size=0.1, train_size=0.9, shuffle=False)

from sktime.forecasting.pytorchforecasting import PytorchForecastingNBeats

model = PytorchForecastingNBeats(
    trainer_params={
        "max_epochs": 5,  # for quick test
        "limit_train_batches": 50,  # for quick test
    },
    dataset_params={
        "max_encoder_length": 3,
    },
)

from sktime.forecasting.base import ForecastingHorizon

fh = ForecastingHorizon([1, 2, 3], is_relative=True)
model.fit(y=y_train, fh=fh)

y_pred = model.predict(y=y_test, fh=fh)
y_pred
```

**Expected behavior**
There would be values in future predictions

**Additional context**

Works with: sktime==0.35.1 torch==2.4.0
Works with: sktime==0.31.0 torch==2.4.0
Does not work: sktime>0.35.1 torch==2.4.0

So, there must be a breaking change in sktime above 0.35.1

**Versions**
<details>

```bash
System:
    python: 3.10.16 | packaged by conda-forge | (main, Dec  5 2024, 14:16:10) [GCC 13.3.0]
executable: /root/work/apps/miniforge3/envs/ai/bin/python
   machine: Linux-6.8.0-53-generic-x86_64-with-glibc2.39

Python dependencies:
          pip: 25.0.1
       sktime: 0.37.0
      sklearn: 1.5.2
       skbase: 0.12.2
        numpy: 1.26.4
        scipy: 1.15.2
       pandas: 2.2.2
   matplotlib: 3.10.1
       joblib: 1.4.2
        numba: 0.61.0
  statsmodels: 0.14.4
     pmdarima: 2.0.4
statsforecast: 1.7.8
      tsfresh: 0.21.0
      tslearn: 0.6.3
        torch: 2.7.0
   tensorflow: 2.19.0
```

**Comment by jobs-git** [comment]
Not a bug but a data set issue, solution: test another data set and make sure `loss` is acceptable
